lib/mysql_framework/scripts/lock_manager.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+require 'redlock'
+
+module MysqlFramework
+  module Scripts
+    class LockManager
+      def initialize
+        @pool = Queue.new
+      end
+
+      def fetch_client
+        @pool.pop(true)
+      rescue StandardError
+        # By not letting redlock retry we will rely on the retry that happens in this class
+        Redlock::Client.new([redis_url], retry_jitter: retry_jitter, retry_count: 1, retry_delay: 0)
+      end
+
+      def with
```

Comment:
should this be a private method?  Or if intentionally public maybe rename it to `with_client` since you already have a `with_lock`.